crates/sovereign-sdk/full-node/db/sov-db/src/ledger_db/tests.rs

```diff
@@ -0,0 +1,122 @@
+use std::sync::OnceLock;
+
+use anyhow::anyhow;
+use sov_schema_db::SchemaBatch;
+
+use super::migrations::{LedgerDBMigrator, LedgerMigration, MigrationName, MigrationVersion};
+use super::LedgerDB;
+use crate::ledger_db::{SharedLedgerOps, TestLedgerOps};
+use crate::rocks_db_config::RocksdbConfig;
+use crate::schema::tables::TestTableOld;
+
+pub fn successful_migrations() -> &'static Vec<Box<dyn LedgerMigration + Send + Sync + 'static>> {
+    static MIGRATIONS: OnceLock<Vec<Box<dyn LedgerMigration + Send + Sync + 'static>>> =
+        OnceLock::new();
+    MIGRATIONS.get_or_init(|| vec![Box::new(OldToNewMigration {})])
+}
+
+pub fn failed_migrations() -> &'static Vec<Box<dyn LedgerMigration + Send + Sync + 'static>> {
+    static MIGRATIONS: OnceLock<Vec<Box<dyn LedgerMigration + Send + Sync + 'static>>> =
+        OnceLock::new();
+    MIGRATIONS.get_or_init(|| vec![Box::new(FailedOldToNewMigration {})])
+}
+
+struct OldToNewMigration {}
+impl LedgerMigration for OldToNewMigration {
+    fn identifier(&self) -> (MigrationName, MigrationVersion) {
+        ("OldToNew".to_owned(), 1)
+    }
+
+    fn execute(&self, ledger_db: sov_rollup_interface::RefCount<LedgerDB>) -> anyhow::Result<()> {
+        let Some(values) = ledger_db.db.get::<TestTableOld>(&())? else {
+            return Ok(());
+        };
+        for (index, value) in values.into_iter().enumerate() {
+            ledger_db.put_value(index as u64, (index as u64, value))?;
+        }
```

Comment:
we could also create the new table here right?

Reply:
Yes, `TestTableNew` is a new table and the call to `put_value` here is adding entries into this new table.

Reply:
but isn't that table already in the schema when you start the process? I'm talking about creating a new table that we haven't defined in `LEDGER_TABLES` previously, but we do now, with the migration

Reply:
So tables are not "created" when you start the process. They are created as soon as you write to them. 

When we create the `inner: DB` part of the `LedgerDB`, we pass the schema of those tables but it won't actually create them until they're written to.

Reply:
The idea behind the `TestTableNew` is to demonstrate how we can define the new table's schema and we add it to the code and we get the code to use it. However, it doesn't mean that it has data to be used. That's what migrations are for.